File: Mobile Android/EsemkaLibrary.Api/EsemkaLibrary.Api/Models/Book.cs
using System;
using System.Collections.Generic;

namespace EsemkaLibrary.Api.Models;

public partial class Book
{
    public int Id { get; set; }

    public string? Title { get; set; }

    public string? Author { get; set; }

    public int? PublicationYear { get; set; }

    public string? Isbn { get; set; }

    public string? Cover { get; set; }

    public virtual ICollection<BookContent> BookContents { get; set; } = new List<BookContent>();

    public virtual ICollection<BookLike> BookLikes { get; set; } = new List<BookLike>();

    public virtual ICollection<Bookmark> Bookmarks { get; set; } = new List<Bookmark>();

    public virtual ICollection<BookCategory> Categories { get; set; } = new List<BookCategory>();
}

File: Mobile Android/EsemkaLibrary/app/src/main/java/com/example/esemkalibrary/Book.java
package com.example.esemkalibrary;

import com.google.gson.annotations.SerializedName;

import java.util.List;

public class Book {
    private int id;
    private String title;
    private String author;
    @SerializedName("publicationYear")
    private int publicationYear;
    private String isbn;
    private String cover;
    @SerializedName("bookContents")
    private List<BookContent> bookContents;

    // Buatlah constructor, getter, dan setter sesuai kebutuhan Anda


    public Book(int id, String title, String author, int publicationYear, String isbn, String cover, List<BookContent> bookContents) {
        this.id = id;
        this.title = title;
        this.author = author;
        this.publicationYear = publicationYear;
        this.isbn = isbn;
        this.cover = cover;
        this.bookContents = bookContents;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getAuthor() {
        return author;
    }

    public void setAuthor(String author) {
        this.author = author;
    }

    public int getPublicationYear() {
        return publicationYear;
    }

    public void setPublicationYear(int publicationYear) {
        this.publicationYear = publicationYear;
    }

    public String getIsbn() {
        return isbn;
    }

    public void setIsbn(String isbn) {
        this.isbn = isbn;
    }

    public String getCover() {
        return cover;
    }

    public void setCover(String cover) {
        this.cover = cover;
    }

    public List<BookContent> getBookContents() {
        return bookContents;
    }

    public void setBookContents(List<BookContent> bookContents) {
        this.bookContents = bookContents;
    }
}
